### PR TITLE
Chore: Ignore errors on Performance Server

### LINKED ISSUE #686: Scala 3.1.0, RC19, throws netty warnings
**Describe the bug**
Every request throws a warning:
```

Dec 23, 2021 4:07:56 PM io.netty.channel.DefaultChannelPipeline onUnhandledInboundException
WARNING: An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer

```

**To Reproduce**
Repro repo: https://github.com/bit-void/zio-http-fail

### PR DESCRIPTION
**Pull request by gciuloaica**
…ome error handler. current implementation is calling the error handler in case of exceptions in the server side.

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/service/Handler.scala [hunk 1]
```diff
@@ -44,6 +44,10 @@ private[zhttp] final case class Handler[R](
     )
   }
 
+  override def exceptionCaught(ctx: Ctx, cause: Throwable): Unit = {
+    config.error.fold(())(f => runtime.unsafeRun(ctx)(f(cause)))
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/service/Handler.scala]: if the errorHandler is not provided I feel we should call `super.exeptionCaught`?

File: zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala [hunk 2]
```diff
@@ -36,7 +36,11 @@ abstract class HttpRunnableSpec(port: Int) extends DefaultRunnableSpec { self =>
   def serve[R <: Has[_]](
     app: HttpApp[R, Throwable],
   ): ZManaged[R with EventLoopGroup with ServerChannelFactory, Nothing, Unit] =
-    Server.make(Server.app(app) ++ Server.port(port) ++ Server.paranoidLeakDetection).orDie
+    Server
+      .make(
+        Server.app(app) ++ Server.port(port) ++ Server.error(err => UIO(println(err))) ++ Server.paranoidLeakDetection,
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala]: This change won't be required if the previous comment is fixed.

**Review by amitksingh1490 — CHANGES_REQUESTED** [review]

File: example/src/main/scala/example/HelloWorldAdvanced.scala [hunk 3]
```diff
@@ -21,10 +21,13 @@ object HelloWorldAdvanced extends App {
     case Method.GET -> !! / "utc"    => clock.currentDateTime.map(s => Response.text(s.toString))
   }
 
+  private val errorHandler: Throwable => ZIO[Any, Nothing, Unit] = cause => UIO(println(cause))
```
Comment (amitksingh1490) [inline example/src/main/scala/example/HelloWorldAdvanced.scala]: We should never use `UIO(println(cause)`
Also we should not update example for minor changes

File: zio-http/src/main/scala/zhttp/service/Handler.scala (hunk 1 above)
Comment (gciuloaica) [reply zio-http/src/main/scala/zhttp/service/Handler.scala]: right. done

File: example/src/main/scala/example/HelloWorldAdvanced.scala (hunk 3 above)
Comment (gciuloaica) [reply example/src/main/scala/example/HelloWorldAdvanced.scala]: ok, done

File: zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala (hunk 2 above)
Comment (gciuloaica) [reply zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala]: done

File: zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala [hunk 4]
```diff
@@ -36,7 +36,11 @@ abstract class HttpRunnableSpec(port: Int) extends DefaultRunnableSpec { self =>
   def serve[R <: Has[_]](
     app: HttpApp[R, Throwable],
   ): ZManaged[R with EventLoopGroup with ServerChannelFactory, Nothing, Unit] =
-    Server.make(Server.app(app) ++ Server.port(port) ++ Server.paranoidLeakDetection).orDie
+    Server
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala]: Unnecessary Change

File: zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala [hunk 5]
```diff
@@ -10,8 +10,8 @@ import zhttp.internal.AppCollection.HttpEnv
 import zhttp.internal.HttpRunnableSpec.HttpIO
 import zhttp.service._
 import zhttp.service.client.ClientSSLHandler.ClientSSLOptions
+import zio._
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala]: Unnecessary Change

Comment (gciuloaica) [reply zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala]: The change was done by scalafmt ...

File: zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala (hunk 4 above)
Comment (gciuloaica) [reply zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala]: The change was done by scalafmt ...

File: example/src/main/scala/example/PlainTextBenchmarkServer.scala [hunk 6]
```diff
@@ -33,7 +33,9 @@ object Main extends App {
   private def server(response: Response[Any, Nothing]) =
     Server.app(app(response)) ++
       Server.port(8080) ++
+      Server.error(_ => UIO.unit) ++
       Server.keepAlive ++
-      Server.disableLeakDetection ++ Server.consolidateFlush
+      Server.disableLeakDetection ++
+      Server.consolidateFlush
```
Comment (tusharmath) [inline example/src/main/scala/example/PlainTextBenchmarkServer.scala:39]: Example doesn't need changing.

File: zio-http/src/main/scala/zhttp/service/Server.scala [hunk 7]
```diff
@@ -42,8 +42,6 @@ object Server {
   private[zhttp] final case class Config[-R, +E](
     leakDetectionLevel: LeakDetectionLevel = LeakDetectionLevel.SIMPLE,
     maxRequestSize: Int = 4 * 1024, // 4 kilo bytes
-
-    // TODO: add error handler
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/Server.scala:46]: Thanks!

File: zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala (hunk 5 above)
Comment (tusharmath) [reply zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala]: I think it's intellij :)

File: example/src/main/scala/example/PlainTextBenchmarkServer.scala (hunk 6 above)
Comment (amitksingh1490) [reply example/src/main/scala/example/PlainTextBenchmarkServer.scala:39]: This is for benchmark @tusharmath, this will make benchmark for non forks robust. Also we wanted to check there is no performance penalty.

Comment (gciuloaica) [reply example/src/main/scala/example/PlainTextBenchmarkServer.scala:39]: Changed to be used during benchmarking phase

Comment (tusharmath) [reply example/src/main/scala/example/PlainTextBenchmarkServer.scala:39]: I would be interested in knowning how #606 will fix it.

**Review by tusharmath — APPROVED** [review]

**Review by amitksingh1490 — APPROVED** [review]
